### DISCUSSION TITLE
ZIP/alpha/correlation/random effects

### DISCUSSION THREAD

**Discussion by AlainZuur** [JenniNiku/gllvm#137] [General]
Hello,
I know that Mr Stork is flying around (or has just been around), hence take your time to respond. 

I have four (hopefully) short questions.

1. ZIP and ZINB. I assume that gllvm can only do a Pi_j that is constant? It can change by species, but you can't model it as a function of covariates..correct? Not that I want to model Pi as a function of covariates (too complicated).......but I just want to confirm that there is no ziformula somewhere that I missed.

2. The alpha in ordiplot.gllvm. That is just a scaling factor to map the site scores and loadings closer to each other? Somehow the words 'biplot' and 'alpha' confused me a little bit. These are not biplots as in PCA, where you typically choose between alpha = 0, alpha = 0.5, or alpha = 1. The 'plot' inside 'ordiplot'...that is the 'plot' from vegan?

3. And one more:
https://jenniniku.github.io/gllvm/articles/vignette1.html
Under: Studying co-occurrence patterns
The 𝚪𝚪...that is being used to get a covariance matrix and then a correlation matrix via getResidualCor. Then there is this sentence:  "The getResidualCor function can be used to estimate the correlation matrix of the linear predictor across species."....so..what you now have is a correlation matrix on the scale of the predictor. But that is not correlation between species due to the link function. But those nice corrplot graphs seem to indicate that it is correlation between species. What am I missing here? I am missing an expression for the covariance/correlation cov(Y_ij, Y_ip).

4. I can't do two different random effects in row.eff?

Nice package! Keep up the good work!

Kind regards,
Alain

**Comment by BertvanderVeen** [comment]
Hello Alain.

1. Currently this is the case, but we are working on allowing effects (fixed and random) on the inflated component in the near future
2. alpha is the scaling factor for a bi/tri/...plot. Ordiplot is custom code written by the gllvm devs and has nothing to do with the vegan R-package
3. In most/many cases there is no solution to the covariance/correlation on the response scale. This is the same as in mixed-effects models which is why (e.g.,) the Nakagawa and Shielzeth R² makes use of (IIRC) a taylor series expansion and/or Jensen's inequality
4. No, not at present. I (we?) hope to expand this in the future (nice to have/low hanging fruit)

Bert
